compiler/optimizer/RegisterCandidate.cpp

```diff
@@ -3758,7 +3758,7 @@ TR_RegisterCandidates::computeAvailableRegisters(TR_RegisterCandidate *rc, int32
             overlaps = overlapLookup->second;
             }
 
-         TR_RegisterCandidate *rc2 = b->getGlobalRegisters(comp())[i].getRegisterCandidateOnEntry();
+         TR_RegisterCandidate *rc2 = b->getGlobalRegisters(comp())[i].getRegisterCandidateOnExit();
```

Comment:
Good catch. Is this a recent bug related to the compile time improvements OR was it just a pre-existing functional bug you are fixing now ?

Reply:
This was a pre-existing functional bug I encountered when trying to reduce the compile time in `computeAvailableRegisters`, using the `GlobalSet` to shortcut some of the checks. The reduction was not significant enough to warrant the additional complexity.